lib/Optimizer/Transforms/LiftArrayAlloc.cpp

```diff
@@ -89,12 +91,16 @@ class AllocaPattern : public OpRewritePattern<cudaq::cc::AllocaOp> {
         cannotEraseAlloc = isLive = true;
       } else {
         for (auto *useuser : user->getUsers()) {
+          if (!useuser)
+            continue;
           if (auto load = dyn_cast<cudaq::cc::LoadOp>(useuser)) {
             rewriter.setInsertionPointAfter(useuser);
             LLVM_DEBUG(llvm::dbgs() << "replaced load\n");
-            rewriter.replaceOpWithNewOp<cudaq::cc::ExtractValueOp>(
-                load, eleTy, conArr,
+            auto extractValue = rewriter.create<cudaq::cc::ExtractValueOp>(
+                loc, eleTy, conArr,
                 ArrayRef<cudaq::cc::ExtractValueArg>{offset});
+            rewriter.replaceAllUsesWith(load, extractValue);
+            insertOpToErase(load);
```

Comment:
I don't see why the `replaceOpWithNewOp` needs to be changed here. It will

  - create the new ExtractValueOp
  - replace all the users of the original LoadOp to use the new ExtractValueOp
 
 which is exactly what your change does. Additionally, we don't need or want to aggressively delete Ops. For example, passes may not be successful, and the MLIR infrastructure is designed to allow it to cleanly back out of the changes without leaving the IR broken. We should just let MLIR do the cleanups anywhere that it can. In this case, a LoadOp with 0 users will simply be removed by the next DCE. We don't have to do anything special.

Reply:
Not sure why but the original code crashes on uccsd example

Reply:
I removed this change for now, let's see if it happens again when all the uccsd changes come together and I might get a repro